### ISSUE TITLE
Issue #150 120" Tall Pards Need more info

### ISSUE THREAD

**Issue by ClassyTeddy** [CCDevNelson/Snap-Updates#54]
Ryan has asked for more info on the 120" Tall pards issue posted in SNaP-Dev

CCDevNelson/SNaP-Dev#150

Neither me nor Ryan know what materials come in 10', and to translate that to SNaP we will most likely need a way to put that information into the .csv at the very least. 

As for how we are going to implement this, how would any of you like for this feature to work? Should it come up with an error message saying that the material is not available in that size? Or should there be a new material tab that is only used for 120" partitions. 

@meltnaz Your info on this will be especially helpful.

**Comment by TEDSTANDAGE** [comment]
This could get tricky as only some material comes in 10', some in 9' (CL and SALT material, not even noted on the csv), some in 8' or 10' depending if we special order it (and we charge more if we special order 10' material for them (ItemTypeCode 15225), so pricing needs to be factored into this), and some colors 3/4" can be ordered in 10' while the corresponding 1/4" might only come in 8' (my understanding is that all 1/4" material is limited to 8' except CL and SALT material which comes in the same sheet size as their corresponding 3/4" material--this would mean that for those colors that only come in 8' 1/4" material that the full backs would be limited to the size the material comes in). I know that's confusing but that's what we are dealing with and this is going to take more thought, and perhaps some reworking of the ItemTypeCodes and making more duplicates of colors with new SKU numbers.

Issues will arise if we generally allow for 10' partitions/backs/etc. and someone tries to send a job through with a material that does not come in 10', and also if the job isn't accurately priced for special ordered 10' material. 

We need to take our time thinking through this one and maybe we can discuss it at our SNaP meeting next week.

**Comment by ClassyTeddy** [comment]
That would probably be good. I will let Ryan know that information regarding this issue may take some time.

**Comment by TEDSTANDAGE** [comment]
From our meeting yesterday and other notes:

-Currently SNaP allows up to 76H partitions (96.50). Generally we want to increase that and allow for partitions up to 120" (94H 119.18), but we also want designers/users to be trained, knowledgeable, and aware of the limitations of the materials they are selling. 

-To help designers/users, we will create a new material drop down list called Oversized Material (adding to the options of Melamine, Veneer, and Textured Melamine). In that list we will have Oxford White, Cabinet Almond, Cleaf, SALT, veneer, and MDF because those materials are stocked locally/standard to get oversized (I spoke with Sean Wilson about this). The display name for those colors should show the max length of that material (e.g. "Oxford White (Max 120")", "S Canyon Bleach (Max 109")", etc.). We may or may not have to create new SKU numbers for CL/ SALT/Veneer/MDF to get them to show up on two separate materials drop downs (already have a separate SKU for white/almond 10' material), but we will wait on Teddy/Ryan to tell us how we will be able to make this happen per Teddy's recent question to Ryan. We may want to have a new SKU number anyway to allow for the different Display Name in the Oversize Material drop down to show the max length, but it would be nice not to have to create a new SKU since materials like CL and SALT come oversized anyway.

-If designers/users want to special order oversized sheets of colors not on that list they need to know they have to call ahead to get a lead time for us to get the material before scheduling the job.

-Is it possible to add a note to the 76H partition size display to say "Oversized Material only" so that users don't try to order 96.50" partition in our standard 8' materials? COS currently doesn't even allow for 96.50" partitions anyway, so would it be best to just not allow for this partition size in SNaP either?

![image](https://user-images.githubusercontent.com/61025894/81431703-72d39b80-9116-11ea-82b8-eadc534c3c62.png)

**Comment by RyanMDev** [comment]
Branch: `120-Tall-Pards`

**Comment by RyanMDev** [comment]
https://github.com/CCDevNelson/SNaP-Dev/issues/150#issuecomment-640886047

**Comment by TEDSTANDAGE** [comment]
I like this. Still would like the max height (in inches) put in the display name, so we may have to update the csv (not sure if we need to do that both in the CCItems csv and this new oversized material csv? May need to create a new SKU number).

The program crashes when I select the following materials and put in the initial partition (the other colors work fine--even CL Cool Grey Linen, and if I start with a working material I can change the color to these crashing materials after I insert a partition, but when these are selected before putting in the initial partition the program crashes when I put in that initial partition--or I haven't waited long enough for it to load but I have waited over a minute or two):

-Cabinet Almond
-CL Rustic Oak
-CL Rialto
-CL Lakeshore Oak
-CL Khaki Linen

We should remove "Cedar Aromatic Ven250 G1" as an option in the oversized material--it is a 1/4" material that we only use as backing material. We just want 3/4" veneer and MDF material in there

We will need to update the max length of CL material to be the same as Salt material.

**Comment by ClassyTeddy** [comment]
Do you get an error message when it crashes? Also, what is MDF material?

**Comment by TEDSTANDAGE** [comment]
No, the mouse just does the circle thing and it sits there thinking:

![image](https://user-images.githubusercontent.com/61025894/84182361-20500c80-aa3f-11ea-8b20-ca36ee332ee8.png)

MDF is the standard base material we use for painted products, alder veneer is our standard base material for stained products.

**Comment by TEDSTANDAGE** [comment]
So I waited about 8 minutes and it finally came through, but it came through as white, not CL Khaki Linen, though I re-selected CL Khaki Linen as my color choice and the correct color showed up. Here is what showed up after waiting a few minutes:

![image](https://user-images.githubusercontent.com/61025894/84183313-8f7a3080-aa40-11ea-92aa-7c16e7f7ab8b.png)

Then after re-selecting CL Khaki Linen:

![image](https://user-images.githubusercontent.com/61025894/84183320-930db780-aa40-11ea-920e-7fa866906504.png)

**Comment by RyanMDev** [comment]
- > Still would like the max height (in inches) put in the display name, so we may have to update the csv (not sure if we need to do that both in the CCItems csv and this new oversized material csv? May need to create a new SKU number).

    I will need to add this in code.

    The `DisplayName` from `OversizeMaterialColors.csv` needs to follow the same rule as `DisplayName` from `CCItems.csv` (match the name of a rendering material .blend file). Furthermore, the `DisplayName` in `OversizeMaterialColors.csv` is used to find the matching regular sized material for the SKU to be exported.

    This is done to avoid having to add new skus for oversized materials in `CCItems.csv` and instead define which materials are available in OS by `DisplayName.`

I will update these but for future reference, the following can all be done in `OversizeMaterialColors.csv` and is as simple as copy/pasting the `DisplayName` from `CCItems.csv` and providing a `MaxLength`:
- Remove Cedar Aromatic Ven250 G1
- Add MDF
Which MDF sku are we looking to use here?
![mdf_skus](https://user-images.githubusercontent.com/6529544/84179665-25ab5800-aa3b-11ea-9555-011554831f8a.png)

- > We will need to update the max length of CL material to be the same as Salt material.

**Comment by RyanMDev** [comment]
Looks like it is taking too long to find the rendering
 material, I will take a look.

**Comment by RyanMDev** [comment]
> Then after re-selecting CL Khaki Linen:

After re-selecting CL Khaki Linen does it change the material from white to CL Khaki Linen like normal? or does it again hang?

**Comment by TEDSTANDAGE** [comment]
After re-selecting it went fairly quick like normal

**Comment by RyanMDev** [comment]
Okay

Try to copy the render material file: `CL Khaki Linen.blend`

From: `SNaP\2.78\scripts\addons\snap_db\materials`

To: `SNaP\data\materials\Cabinet Materials\Classy Closets`

Leave the original in place and see if that  fixes the problem.

**Comment by TEDSTANDAGE** [comment]
> Okay
> 
> Try to copy the render material file: `CL Khaki Linen.blend`
> 
> From: `SNaP\2.78\scripts\addons\snap_db\materials`
> 
> To: `SNaP\data\materials\Cabinet Materials\Classy Closets`
> 
> Leave the original in place and see if that fixes the problem.

That fixed the issue

**Comment by RyanMDev** [comment]
Commit: CCDevNelson/snap-dev@b000a49498bc32585e4a903fece36a61a45950ca

Adds max length label to UI:
![max_len_1](https://user-images.githubusercontent.com/6529544/84190764-4b8d2880-aa4c-11ea-92bd-9499d930e593.png)
![max_len_2](https://user-images.githubusercontent.com/6529544/84190773-4f20af80-aa4c-11ea-8492-1cdaf2e8d05e.png)

Commit: CCDevNelson/snap-dev@65e852fafb0b526dc375d8ea4bbcb29ae97799c5

Updates `OversizeMaterialColors.csv`
- Remove 'Cedar Aromatic Ven250 G1'
- Add 'MDF 4X10-3/4'
- Set max length of CL materials to match SALT materials

**Comment by RyanMDev** [comment]
Currently there is no rendering material for 'MDF' so until we have one in place MDF will not appear in the list.

**Comment by RyanMDev** [comment]
There is a 'MDF' material in the standard material library, but it is unfinished.
![mdf](https://user-images.githubusercontent.com/6529544/84191677-b723c580-aa4d-11ea-8387-3e61ce32e6ca.png)

Is the MDF painted, stained?

**Comment by TEDSTANDAGE** [comment]
MDF is our standard base material we apply paint to

**Comment by RyanMDev** [comment]
Alright, sounds like MDF either needs to be it's own material type with a selection of the available paint colors.

Do any of these paint colors match any existing render materials?

Or do we need to add new colors for paint?
